packages/react-error-overlay/src/index.js

```diff
@@ -167,3 +167,24 @@ window.__REACT_ERROR_OVERLAY_GLOBAL_HOOK__.iframeReady = function iframeReady()
   isLoadingIframe = false;
   updateIframeContent();
 };
+
+const warningText =
+  'When deploying an application, `react-error-overlay` should be excluded ' +
+  'as it is a heavy dependency meant for development.\n\n' +
+  'Consider adding an error boundary to your tree to customize error ' +
+  'handling behavior. See https://fb.me/react-error-boundaries for more ' +
+  'information.';
+
+if (process.env.NODE_ENV !== 'production') {
+  var testFunc = function testFn() {};
+  if ((testFunc.name || testFunc.toString()).indexOf('testFn') === -1) {
```

Comment:
`toString()` is not guaranteed to be safe. I'd opt to remove this branch completely and for now just assume `process.env.NODE_ENV` is correct. We have detection in React DevTools for other cases.